src/statistic/statistic.js

```diff
@@ -0,0 +1,131 @@
+     * @param aQuestion an object Question that is a Question contained in the database
+     * @returns {{title: *, numberVotes: *}} array containing title: the title of the question and numberVotes: the total
+     * of all the votes for the responses for this question
+     */
+    questionFormat(aQuestion){
+        logger.debug(`Formatting question : ${aQuestion}`);
+        return {
+            title: aQuestion.titreQuestion,
+            numberVotes: aQuestion.numberVotes
+        }
+    }
+
+    /**
+     * Will format the response to have only useful data for statistics
+     * @param aResponse an object Response that is a Response contained in the database
+     * @param numberTotalVotes the number of votes for the response
+     * @returns {{response: *, percentage: number, numberVotes: *}}
+     */
+    responseFormat(aResponse, numberTotalVotes){
+        return {
+            response: aResponse.response,
+            numberVotes: aResponse.devices.length,
+            percentage: Math.round((aResponse.devices.length/numberTotalVotes) * 100)
+        }
+    }
+
+    /**
+     * Return all the informations desired when we want to make stats for an admin
+     * @param aUsername String that represents the username of the admin
+     * @returns {Promise<*[]>} an Array containing all the stats of the admin
+     */
+    async adminStats(aUsername){
+        let allDiscussions = await dbManager.getDiscussionsAdmin(aUsername);
+
+        return [allDiscussions.length, Array.from(allDiscussions.values(), d => this.discussionFormat(d))];
+    };
+
+    /**
+     * Get the number of the total of votes for all the responses for a question.
+     * @param aQuestionId integer that is the id of the question
+     * @param aDiscussionId integer that is the id of the discussion
+     * @returns {Promise<number>} an integer that is the number of votes
+     */
+    async getNumberVotesQuestion(aQuestionId, aDiscussionId) {
+        // Get all the responses related to the question desired
+        let allResponses = await dbManager.getResponsesQuestion(aQuestionId, aDiscussionId);
+        let numberTotalVotes = 0;
+        // Get all the votes for all the responses
+        for (let i = 0; i < allResponses.length; ++i) {
+            numberTotalVotes += allResponses[i].devices.length;
+        }
+        logger.info(`The number of votes : ${numberTotalVotes}`);
+        return numberTotalVotes;
+    }
+
+    /**
+     * Get all the stats for a debate
+     * @param aDiscussionId the id of the discussion to get the stats from
+     * @returns {Promise<*[]>} an array containing the stats wanted for a debate
+     */
+    async debateStats(aDiscussionId){
+        // Get the discussion in the database
+        let debate = await dbManager.getDiscussion(aDiscussionId);
+        // Get all the questions related to the debate
+        let allQuestions = await dbManager.getQuestionsDiscussion(aDiscussionId);
```

Comment:
Same here. Can `dbManager.getDiscussion` or `dbManager.getQuestionsDiscussion` return a null object ?

Reply:
As said before corrected in the new commit